lib/services/elasticsearch.js

```diff
@@ -153,718 +168,635 @@ class ElasticSearch extends Service {
-
-    esRequest.body = request.input.body;
-    delete esRequest.body.includeTrash;
+  mGet (index, collection, ids) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: ids
+    };
 
     return this.client.mget(esRequest)
-      .then(result => {
-        // harmonize response format based upon the search one
-        if (result.docs) {
-          result.hits = request.input.args.includeTrash
-            ? result.docs
-            : result.docs.filter(
-              doc => doc._kuzzle_info === undefined ||
-              doc._kuzzle_info.active
-            );
-          delete result.docs;
-
-          result.hits = result.hits.map(obj => prepareMetadata(obj));
-        }
-
-        return result;
-      })
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .then(({ body }) => ({
+        hits: body.docs,
+        total: body.total
+      }))
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Count how many documents match the filter given in body
-   * @param {Request} request
-   * @returns {Promise} resolve the number of document
+   * Counts how many documents match the filter given in body
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} query - Query to match
+   *
+   * @returns {Promise.<Object>} { count }
    */
-  count(request) {
-    const esRequest = initESRequest(request);
-
-    esRequest.body = request.input.body;
+  count (index, collection, query = {}) {
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: query
+    };
 
-    if (!request.input.args.includeTrash) {
-      addActiveFilter(esRequest);
-    }
     return this.client.count(esRequest)
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+      .then(({ body }) => ({
+        count: body.count
+      }))
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
    * Sends the new document to elasticsearch
    * Cleans data to match elasticsearch specifications
    *
-   * @param {Request} request
-   * @returns {Promise} resolve an object that contains _id
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Object} content - Document content
+   * @param {Object} options - id (null), refresh (false), userId (null)
+   *
+   * @returns {Promise.<Object>} { _id, _version, _source }
    */
-  create(request) {
-    const esRequest = initESRequest(request, ['refresh']);
-
-    esRequest.id = request.input.resource._id;
-    esRequest.body = request.input.body;
+  create (
+    index,
+    collection,
+    content,
+    { id=null, refresh=false, userId=null } = {})
+  {
+    assertIsObject(content);
+
+    const esRequest = {
+      index: this._getESIndex(index, collection),
+      body: content,
+      op_type: 'create',
+      id,
+      refresh,
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this trailing comma. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ATrailingComma)